src/app/utils/accountPageUtils.js

```diff
@@ -22,57 +24,162 @@ const makeRequest = (
       if (data.error) updateErrorMessage(data.error);
       return updateAccountHtml(data);
     })
-    .catch(res => console.log('ERROR', res));
+    .catch(res => console.error('ERROR', res));
```

Comment:
There are a couple of things happening which is making the loading skeleton not show up. For one, the `useEffect` in `AccountPage` runs multiple times and so the `toggleLoadingState(false);` in `manipulateAccountPage` keep taking precedence over the call in this function.

You also always want to do a loading state before making an ajax call and then right after, instead of only making the call before and then setting it to "false" _somewhere else_. You can fix this by removing the other call in `manipulateAccountPage` and put a `.finally(() => toggleLoadingState(false));` here on line 28. Then, whether you get a good or bad response from the API, you're guaranteed to stop the loading skeleton.

Reply:
Oh I did not know about `finally`. That's awesome. Thanks!